pageserver/src/layered_repository/timeline.rs

```diff
@@ -849,6 +868,34 @@ impl LayeredTimeline {
         }
     }
 
+    /// Update current logical size, adding `delta' to the old value.
+    fn update_current_logical_size(&self, delta: i64) {
+        let new_size = self
+            .current_logical_size
+            .fetch_add(delta, AtomicOrdering::SeqCst);
+
+        // Also set the value in the prometheus gauge. Note that
+        // there is a race condition here: if this is is called by two
+        // threads concurrently, the prometheus gauge might be set to
+        // one value while current_logical_size is set to the
+        // other. Currently, only initialization and the WAL receiver
+        // updates the logical size, and they don't run concurrently,
+        // so it cannot happen. And even if it did, it wouldn't be
+        // very serious, the metrics would just be slightly off until
+        // the next update.
+        self.current_logical_size_gauge.set(new_size);
```

Comment:
Gauge has `add` method. Will it help to avoid the race?

Reply:
That would fix it for concurrent calls to `update_current_logical_size`, but it would remain or become worse for the case that `set_current_logical_size' and `update_current_logical_size` are called concurrently.

Reply:
Hmm, if we consider lazy loading in the background, then we cannot use set_current_logical_size anyway, we'll need to turn it into update_current_logical_size so it can be added to the value that is calculated from ingested wal